x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -42,9 +78,9 @@ pub extern "C" fn alloc(len: usize) -> *mut u8 {
         std::alloc::handle_alloc_error(layout);
     }
 
-    GLOBAL_STORE.with_borrow_mut(|s| s.insert(ptr, len));
+    ALLOCATIONS.with_borrow_mut(|s| s.insert(ptr, len));
 
-    ptr
+    HostPtr(ptr.cast())
```

Comment:
we could be slightly more explicit with a `cast_const()` here
```suggestion
    HostPtr(ptr.cast_const())
```

Reply:
Good catch